app/assets/javascripts/admin/index_utils/services/pending_changes.js.coffee

```diff
@@ -6,7 +6,7 @@ angular.module("admin.indexUtils").factory "pendingChanges", ($q, resources, Sta
     add: (id, attr, change) =>
       @pendingChanges["#{id}"] = {} unless @pendingChanges.hasOwnProperty("#{id}")
       @pendingChanges["#{id}"]["#{attr}"] = change
```

Comment:
Is there a chance to simplify this algorithm that checks for pending changes? Now, if I'm not mistaken, it's enough with `@pendingChanges` being a boolean to know whether there are changes or not, or even no `@pendingChanges` and simply displaying the message whenever a change is applied.

Reply:
I'm not 100% sure about this (I'm just beginning with Angular as a matter of fact, I'm more comfortable with Rails) but this method doesn't check for pending changes, it adds the pending changes in client side memory (again I'm not quite sure but at least it adds them in the DOM), and then wait for submitAll(form) method to save them.

Could someone with Angular knowledge confirm this ? Or maybe shed some light on the add method ?

Reply:
perhaps @mkllnk can help us here? I understand that @HugsDaniel so up to you. We might have to create an issue out of this though.

Reply:
So the idea is just to display a "you have unsaved changes" message, without the count? That seems like a reasonable solution.

I wouldn't change `@pendingChanges`. It's not a boolean, it's a hash that holds the form changes before they are saved.

Reply:
Yes, a boolean wouldn't know which changes have been applied and which ones are still on the way.

Reply:
:+1: